whois-api/src/main/java/net/ripe/db/whois/api/httpserver/RemoteAddressCustomizer.java

```diff
@@ -1,46 +1,55 @@
 
-import com.google.common.base.Splitter;
-import com.google.common.base.Strings;
-import com.google.common.collect.Iterables;
-import com.google.common.net.HttpHeaders;
+import com.google.common.net.InetAddresses;
+import net.ripe.db.whois.common.ip.Interval;
+import net.ripe.db.whois.common.ip.IpInterval;
+import org.apache.commons.lang.StringUtils;
 import org.eclipse.jetty.server.Connector;
 import org.eclipse.jetty.server.HttpConfiguration;
 import org.eclipse.jetty.server.Request;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
-import javax.annotation.Nullable;
 import java.net.InetSocketAddress;
-import java.util.Enumeration;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 /**
- * When HTTP requests are proxied via a loadbalancer, the X-Forwarded-For value will replace the request address.
- * In case of multiple (comma separated) values in X-Forwarded-For, the last value is used.
+ * When HTTP requests via trusted source use clientIp query paramater as remote Address if set
  */
-public class RemoteAddressCustomizer  implements HttpConfiguration.Customizer {
+public abstract class RemoteAddressCustomizer implements HttpConfiguration.Customizer {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(RemoteAddressCustomizer.class);
+    public static final String QUERY_PARAM_CLIENT_IP = "clientIp";
+    private final Set<Interval> trusted;
 
-    private static final Splitter COMMA_SPLITTER = Splitter.on(',').omitEmptyStrings().trimResults();
+    public RemoteAddressCustomizer(final String trustedIpRanges) {
+        this.trusted = getIntervals(trustedIpRanges.split(","));
+    }
 
     @Override
     public void customize(final Connector connector, final HttpConfiguration httpConfiguration, final Request request) {
-        request.setRemoteAddr(InetSocketAddress.createUnresolved(getRemoteAddress(request), request.getRemotePort()));
+
+        final String remoteAddress = getRemoteAddrForTrustedSource(request, getRemoteAddrForScheme(request));
+        request.setRemoteAddr(InetSocketAddress.createUnresolved(remoteAddress, request.getRemotePort()));
     }
 
-    @Nullable
-    private String getRemoteAddress(final Request request) {
-        final Enumeration<String> headers = request.getHeaders(HttpHeaders.X_FORWARDED_FOR);
-        if (headers == null || !headers.hasMoreElements()) {
-            return request.getRemoteAddr();
-        }
+    private String getRemoteAddrForTrustedSource(final Request request, final String resourceAddr) {
+        final Interval ipResource = IpInterval.asIpInterval(InetAddresses.forString(resourceAddr));
 
-        final String header = headers.nextElement();
-        if (Strings.isNullOrEmpty(header)) {
-            return request.getRemoteAddr();
+        final String clientIp = request.getParameterMap().containsKey(QUERY_PARAM_CLIENT_IP) ? request.getParameter(QUERY_PARAM_CLIENT_IP) : null;
```

Comment:
I have a question related to this changes.
I see that you are using hierarchy, being this class the parent class and RemoteAddresXCustomizer the child.
This change seems that belong to RemoteAddressHttpsCustomizer and not to the parent class.

Should this customize be in the children and RemoteAddressHttpCustomizer do what it is in master branch 
and just apply those changes to RemoteAdressHttpsCustomizer? or RemoteAdressHttpCustomizer need to be affected for this changes as well?

Reply:
reading the story: We need to make sure that the coming IP is a trusted IP
so I guess that this is for both http or https